src/lib/operators/jit_operator/operators/jit_read_tuple.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "jit_abstract_operator.hpp"
+#include "storage/chunk.hpp"
+#include "storage/table.hpp"
+
+namespace opossum {
+
+/* Base class for all column readers.
+ * We need this class, so we can store a number of JitColumnReaders with different template
+ * specializations in a common data structure.
+ */
+class BaseJitColumnReader {
+ public:
+  using Ptr = std::shared_ptr<const BaseJitColumnReader>;
+
+  virtual void read_value(JitRuntimeContext& ctx) const = 0;
+  virtual void increment(JitRuntimeContext& ctx) const = 0;
```

Comment:
Please remember to write comments for the methods. Right now, I have no idea what the difference between these two is.